tests/MyGame/Example/Monster.java

```diff
@@ -344,5 +347,372 @@ public static Monster __lookup_by_key(Monster obj, int vectorLocation, String ke
+    int _test = _o.getTest() == null ? 0 : AnyUnion.pack(builder, _o.getTest());
+    int _test4 = 0;
+    if (_o.getTest4() != null) {
+      startTest4Vector(builder, _o.getTest4().size());
+      for (MyGame.Example.TestT _e : _o.getTest4()) { MyGame.Example.Test.pack(builder, _e);}
+      _test4 = builder.endVector();
+    }
+    int _testarrayofstring = 0;
+    if (_o.getTestarrayofstring() != null) {
+      int[] __testarrayofstring = new int[_o.getTestarrayofstring().size()];
+      int _j = 0;
+      for (String _e : _o.getTestarrayofstring()) { __testarrayofstring[_j] = builder.createString(_e); _j++;}
+      _testarrayofstring = createTestarrayofstringVector(builder, __testarrayofstring);
+    }
+    int _testarrayoftables = 0;
+    if (_o.getTestarrayoftables() != null) {
+      int[] __testarrayoftables = new int[_o.getTestarrayoftables().size()];
+      int _j = 0;
+      for (MyGame.Example.Monster _e : _o.getTestarrayoftables()) { __testarrayoftables[_j] = MyGame.Example.Monster.pack(builder, _e); _j++;}
+      _testarrayoftables = createTestarrayoftablesVector(builder, __testarrayoftables);
+    }
+    int _enemy = _o.getEnemy() == null ? 0 : MyGame.Example.Monster.pack(builder, _o.getEnemy());
+    int _testnestedflatbuffer = 0;
+    if (_o.getTestnestedflatbuffer() != null) {
+      int[] __testnestedflatbuffer = new int[_o.getTestnestedflatbuffer().size()];
+      int _j = 0;
+      for (int _e : _o.getTestnestedflatbuffer()) { __testnestedflatbuffer[_j] = _e; _j++;}
+      _testnestedflatbuffer = createTestnestedflatbufferVector(builder, __testnestedflatbuffer);
+    }
+    int _testempty = _o.getTestempty() == null ? 0 : MyGame.Example.Stat.pack(builder, _o.getTestempty());
+    int _testarrayofbools = 0;
+    if (_o.getTestarrayofbools() != null) {
+      boolean[] __testarrayofbools = new boolean[_o.getTestarrayofbools().size()];
+      int _j = 0;
+      for (boolean _e : _o.getTestarrayofbools()) { __testarrayofbools[_j] = _e; _j++;}
+      _testarrayofbools = createTestarrayofboolsVector(builder, __testarrayofbools);
+    }
+    int _testarrayofstring2 = 0;
+    if (_o.getTestarrayofstring2() != null) {
+      int[] __testarrayofstring2 = new int[_o.getTestarrayofstring2().size()];
+      int _j = 0;
+      for (String _e : _o.getTestarrayofstring2()) { __testarrayofstring2[_j] = builder.createString(_e); _j++;}
+      _testarrayofstring2 = createTestarrayofstring2Vector(builder, __testarrayofstring2);
+    }
+    int _testarrayofsortedstruct = 0;
+    if (_o.getTestarrayofsortedstruct() != null) {
+      startTestarrayofsortedstructVector(builder, _o.getTestarrayofsortedstruct().size());
+      for (MyGame.Example.AbilityT _e : _o.getTestarrayofsortedstruct()) { MyGame.Example.Ability.pack(builder, _e);}
+      _testarrayofsortedstruct = builder.endVector();
+    }
+    int _flex = 0;
+    if (_o.getFlex() != null) {
+      int[] __flex = new int[_o.getFlex().size()];
```

Comment:
why ints? aren't these bytes in the schema? and probably this allocation can be avoided.

Reply:
In the schema these are ubytes which translate to ints in java because there are no unsigned primitives in Java.
I'm not sure if we can't just use bytes, for ubytes, but I think that without the proper context we can get into trouble.
Anyway, this discussion is out of the scope of this feature.

Reply:
I doubled checked it and it seems like we are inconsistent about our behavior when it comes to unsigned values in Java.
When we create a vector of uint16 or uint32 the parameter in the method for the actual data is an array of a larger variable:
uint16 -> we use int,
```
  public static int createUint16Vector(FlatBufferBuilder builder, int[] data) { builder.startVector(2, data.length, 2); for (int i = data.length - 1; i >= 0; i--) builder.addShort((short) data[i]); return builder.endVector(); }
```
uint32 -> we use long.
```
  public static int createUintVectorVector(FlatBufferBuilder builder, long[] data) { builder.startVector(4, data.length, 4); for (int i = data.length - 1; i >= 0; i--) builder.addInt((int) data[i]); return builder.endVector(); }
```
for uint64 java has no native solution and in general it is not commonly used.
Here it becomes interesting: when we create a vector of uint8 the parameter in the method for the actual data is an array of bytes, which natively cannot represent the correct type, but for the purpose of embedding the data into the buffer works good (as we don't simple read it, but type the exact binary format to the buffer).
```
  public static int createUint8VectorVector(FlatBufferBuilder builder, byte[] data) { return builder.createByteVector(data); }
```
The reason for this is the difference in implementation of the function, where bytes has specialize function for creating byte vector.
I think it would be better to decide on one common practice, as if I handle it myself in ObjectAPI, this weird behavior will persist.

Reply:
The inconsistency that you mentioned sounds like a bug and should be addressed separately. Even if it works out at the end.

On Java the original behaviour was to promote unsigned scalars to signed with the next biggest width, so I think this behaviour should be kept.

Reply:
> The inconsistency that you mentioned sounds like a bug and should be addressed separately. Even if it works out at the end.
> 
> On Java the original behavior was to promote unsigned scalars to signed with the next biggest width, so I think this behavior should be kept.

I just afraid we can cause backward compatibility issues while fixing it. I suggest the following:
1. Let's keep the behavior as the current "wrong" implementation, and I will fix this PR to support this inconsistent behavior.
2. I will open a new issue regarding this behavior, where we can discuss what should be the solution for it, and after this PR is merged, I will fix this issue too.

Reply:
Yes, we use the "1 size bigger" trick for unsigned, and we should keep doing this. I would expect `short` used here though.

Reply:
You can see here that `short` is not in use because of casting considerations.
https://github.com/google/flatbuffers/blob/21b7044182b2f700d08f3ff82f8d943738d3b4e5/src/idl_gen_java.cpp#L204
I actually agree with that decision.

Reply:
That makes a lot of sense for a single transient `int` returned from an API, but for the object API where you store possibly whole arrays of these scalars permanently, the memory savings of using `short` might be worth it over the casting.  In fact, for large arrays it may even be desirable for a FlatBuffer `ubyte` to be stored in a Java `byte`. @paulovap opinion?

Reply:
Let's continue this discuss in the other issue I opened.
https://github.com/google/flatbuffers/issues/6574

For now I handled specifically this case, so it's no longer blocker for this PR.